.travis.yml

```diff
@@ -0,0 +1,18 @@
+language: java
+
+jdk:
+  - oraclejdk8
+
+notifications:
+  email: false
+  
+# whitelist
+branches:
+  only:
+    - trilok_travis_ci
+    - master
+    
+sudo: false 
+
+script: 
+  ./gradlew product
```

Comment:
This should be to run all the unit tests: ./gradlew precheckin -Pstore

Reply:
To save time, this one can skip store tests: ./gradlew precheckin, while there can be a separate configuration for PR merges in snappy-store repository that can run with -Pstore

Reply:
The product target takes about 30 minutes to run on Travis's (free, with shared infrastructure for machines/ VMs) environment. Precheckin even escluding store tests will run for several hours (more than 4 hours without parallel tests), although we could try to optimise that by caching etc.  

I get that the point of PR build is to run tests though. Shall we identify a smaller smoke tests suite for that purpose? 

Branch builds can be scheduled with chron jobs to run less frequently. We can run the full test suites and build the product there.

Reply:
Build is triggered on Travis site using GitHub notifications and cannot be triggered from the command line. Travis supports environment variables though. The Travis command line client can trigger a GitHub repo sync.